### PR TITLE
[RCV2 Linux] Support nested directories for output and error blobs

### PR DESCRIPTION
**Pull request by vivlingaiah**
- Support writing to blobs under nested directories under container like for ex. https://mystorageaccount.blob.core.windows.net/mycontainer/dir2/dir3/outputL.txt
- This fixes bug https://msazure.visualstudio.com/One/_workitems/edit/15173484

### CODE REVIEW

File: pkg/download/blob.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ func GetSASBlob(blobURI, blobSas, targetDir string) (string, error) {
 		return "", errors.Wrapf(err, "unable to open storage container: %q", blobURI)
 	}
 
-	// Extract the file name only
-	fileName := getFileName(blobURI)
+	// Extract the llob path after container name
```
Comment (norakoiralamsft) [inline pkg/download/blob.go:81]: v nit: spelling of blob

**Review by norakoiralamsft — APPROVED** [review]

File: pkg/download/blob.go [hunk 2]
```diff
@@ -121,25 +121,35 @@ func CreateAppendBlob(blobURI, blobSas string) (*storage.Blob, error) {
...
 
 	return blobref, nil
 }
 
-// Extract the file name only from the blob uri
-func getFileName(blobURI string) string {
-	s := strings.Split(blobURI, "/")
-	if len(s) > 0 {
-		return s[len(s)-1]
+// Extract the suffix after the container name from blob uri
```
Comment (malhotrarohit) [inline pkg/download/blob.go:138]: consider adding a unit test for this utility method

File: pkg/download/blob.go [hunk 3]
```diff
@@ -121,25 +121,35 @@ func CreateAppendBlob(blobURI, blobSas string) (*storage.Blob, error) {
...
 	}
 
-	fileName := getFileName(blobURI)
+	fileName, blobPathError := getBlobPathAfterContainerName(blobURI, containerRef.Name)
 	if fileName == "" {
-		return nil, fmt.Errorf("cannot extract file name from URL: %q", blobURI)
+		return nil, errors.Wrapf(blobPathError, "cannot extract blob path name from URL: %q", blobURI)
 	}
 
 	blobref := containerRef.GetBlobReference(fileName)
```
Comment (malhotrarohit) [inline pkg/download/blob.go:129]: what would happen if no nested directories were specified inside the container, fileName might be empty in that case right?

File: pkg/download/blob.go (hunk 1 above)
Comment (vivlingaiah) [reply pkg/download/blob.go:81]: I'll address this part of https://msazure.visualstudio.com/One/_workitems/edit/15140524
Need to take this to GA today.

File: pkg/download/blob.go (hunk 3 above)
Comment (vivlingaiah) [reply pkg/download/blob.go:129]: We take blob URL. If no nested directories, blob name would be the "fileName"

File: pkg/download/blob.go (hunk 2 above)
Comment (vivlingaiah) [reply pkg/download/blob.go:138]: I'll address this part of https://msazure.visualstudio.com/One/_workitems/edit/15140524
Need to take this to GA today.

**Pull request merged**